### PR TITLE
Handle single CPU issue requiring detachment before casting to numpy

### PR DESCRIPTION
**Pull request by lucasdurand**
Was running into an error when trying this with only a single CPU. I don't fully understand the ramifications of this change, but it seems to work in a pinch.

### CODE REVIEW

**Comment by rtqichen** [comment]
Oh weird. Is torch.no_grad() not working on CPU?

In any case, I think it's safe to just detach, so can you remove the if part of the line?

Thanks for looking into this.

**Comment by lucasdurand** [comment]
No worries, happy to be the first person to try training this without a GPU
